contrib/keyboard-layout-per-window.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+
+# This script keeps track of active keyboard layouts per window.
+#
+# This script requires i3ipc-python package (install it from a system package
+# manager or pip).
+
+import i3ipc
+
+ipc = i3ipc.Connection()
+prev_focused = ipc.get_tree().find_focused().id
+windows = {}
+
+def on_window_focus(ipc, event):
+    global windows, prev_focused
```

Comment:
Use global is bad practice. Better explicitly pass through args.

Reply:
I agree with you, but here it's part of a callback function call chain. AFAIK, there's no clean way of passing those variables in the callback invocations. So, I think this is a good balance between being simple and readable.